src/ServiceModel/ServiceModel.Proxies/ServiceModel/CommunicationObjectExtensions.cs

```diff
@@ -0,0 +1,113 @@
+{
+    /// <summary>
+    /// A set of extension methods around <see cref="ICommunicationObject"/>.
+    /// </summary>
+    public static class CommunicationObjectExtensions
+    {
+        /// <summary>
+        /// Asynchronously opens the connection of the proxy.
+        /// </summary>
+        /// <exception cref="ArgumentNullException">The <paramref name="communicationObject"/> is null.</exception>
+        public static Task OpenAsync(this ICommunicationObject communicationObject)
+        {
+            _ = communicationObject ?? throw new ArgumentNullException(nameof(communicationObject));
+
+            return Task.Factory.FromAsync(communicationObject.BeginOpen(null, null), communicationObject.EndOpen);
+        }
+
+        /// <summary>
+        /// Asynchronously closes the connection of the proxy.
+        /// </summary>
+        /// <exception cref="ArgumentNullException">The <paramref name="communicationObject"/> is null.</exception>
+        public static Task CloseAsync(this ICommunicationObject communicationObject)
+        {
+            _ = communicationObject ?? throw new ArgumentNullException(nameof(communicationObject));
+
+            return Task.Factory.FromAsync(communicationObject.BeginClose(null, null), communicationObject.EndClose);
+        }
+
+        /// <summary>
+        /// Safely disposes the channel by either closing or aborting it.
+        /// </summary>
+        /// <param name="communicationObject">The channel to dispose.</param>
+        /// <param name="logger">A logger to be used for logging purposes.</param>
+        /// <exception cref="ArgumentNullException">The <paramref name="communicationObject"/> is null.</exception>
+        public static async ValueTask DisposeChannelAsync(this ICommunicationObject communicationObject, ILogger? logger = null)
+        {
+            _ = communicationObject ?? throw new ArgumentNullException(nameof(communicationObject));
+
+            logger ??= NullLogger.Instance;
+
+            try
+            {
+                switch (communicationObject.State)
+                {
+                    case CommunicationState.Opened:
+                        logger.LogDebug("Closing connection");
+                        await communicationObject.CloseAsync().ConfigureAwait(false);
+                        break;
+
+                    case CommunicationState.Faulted:
+                        logger.LogDebug("Aborting connection");
+                        communicationObject.Abort();
+                        break;
+
+                    case CommunicationState.Closed:
+                    case CommunicationState.Closing:
+                    case CommunicationState.Created:
+                    case CommunicationState.Opening:
+                        break;
+
+                    default:
+                        throw new InvalidOperationException($"The channel is in an unexpected state: {communicationObject.State:G}");
```

Comment:
Why the string format 'G' on the enum? Does not look like it has any effect.

```
	$"{Foo.Bar:G}".Dump(); => Bar
	$"{Foo.Bar}".Dump();   => Bar
	$"{Foo.Bar:D}".Dump(); => 0
```

```
public enum Foo
{
	Bar
}
```

Reply:
Suggested by resharper actually.